build/hpc_install.sh

```diff
@@ -0,0 +1,157 @@
+# Cluster specific setup
+if [[ $1 == "longleaf" ]]; then
+    # Setup general purpose user variables needed for Longleaf
+    USERO=$( echo $USER | awk '{ print substr($0, 1, 1) }' )
+    USERN=$( echo $USER | awk '{ print substr($0, 2, 1) }' )
+    USERDIR="/users/$USERO/$USERN/$USER"
+    cd $USERDIR
+
+    # Load required modules
+    module purge
+    module load gcc/9.1.0
+    module load anaconda/2023.03
+    module load git
+elif [[ $1 == "rockfish" ]]; then
+    # Setup general purspose user variables needed for RockFish
+    USERDIR="/scratch4/struelo1/flepimop-code/$USER/"
+
+    # Load required modules
+    module purge
+    module load gcc/9.3.0
+    module load anaconda/2020.07
+    module load git/2.42.0
+else
+    echo "The cluster name '$1' is not recognized, must be one of: 'longleaf', 'rockfish'."
+    exit 1
+fi
+
+# Make sure the credentials is is where we expect and have the right perms
+if [ ! -f "$USERDIR/slack_credentials.sh" ]; then
+    echo "You need to place sensitive credentials in '$USERDIR/slack_credentials.sh'."
+    exit 1
+fi
+chmod 600 $USERDIR/slack_credentials.sh
+source $USERDIR/slack_credentials.sh
+
+# Ensure we have a $FLEPI_PATH
+if [ -z "${FLEPI_PATH}" ]; then
+    echo "An explicit \$FLEPI_PATH was not provided, setting to '$USERDIR/flepiMoP'."
+    export FLEPI_PATH="$USERDIR/flepiMoP"
+fi
+
+# Test that flepiMoP is located there
+if [ ! -d "$FLEPI_PATH" ]; then
+    echo "Did not find flepiMoP at '$FLEPI_PATH', cloning on your behalf."
+    git clone https://github.com/HopkinsIDD/flepiMoP.git $FLEPI_PATH
+elif [ ! -d "$FLEPI_PATH/.git" ]; then
+    echo "The flepiMoP found at '$FLEPI_PATH' is not a git clone, unsure of how to proceed."
+    exit 1
+fi
+
+# Setup the conda environment
+if [ ! -d "$USERDIR/flepimop-env" ]; then
+cat << EOF > $USERDIR/environment.yml
+channels:
+- conda-forge
+- defaults
+dependencies:
+- python=3.10
```

Comment:
As far as I'm aware 3.10 is the only officially supported python version for `gempyor`, see: https://github.com/HopkinsIDD/flepiMoP/blob/427082ce68d68ad8e31c50a537f900aa986b2727/.github/workflows/ci.yml. I've also had great success using 3.11 locally, but that's just local. There are some issues with jumping to 3.12, see: GH-244, maybe we should bump up on priority with 3.13 coming out next week so we can get up to speed with the python ecosystem? I can add 3.11 to our supported versions in either GH-280 or GH-282 if that's of interest, but that's a separate from this PR. In the longer term, our python actions should look more like: https://github.com/ACCIDDA/flepiscripts/blob/initial-package/.github/workflows/ci.yml.

Reply:
From experience with bumping up python versions in software: after a python version is released it will take some time for the devs of your downstream dependencies to port to the new python release. Idk how long this takes, but I lag my software about one year behind the new python version.

Reply:
Love the work with the GitHub action. As it is very painful to get everyone to update their running environment, I would jump ahead and set-up this script on python 3.12 (if the blockers are just deprecation warnings at the moment) or python 3.11 (if worse blocker than that). I realized my rockfish env is 3.12 so I have tested it on this version. 

That only apply if this script is to replace the way one installs on longleaf/rockfish.

Reply:
> As it is very painful to get everyone to update their running environment...

This script should alleviate those issues, and besides, how often do we realistically need to update an environment, once a quarter? That's not very frequent.

I'm happy to bump this up to 3.11 after GH-280 is merged and I'd be willing to do the same for 3.12 in GH-280 (if you would like me to do this then please place a comment there). However, I really don't want us to continue the practice of ignoring warnings so I would not recommend making the default here 3.12. We can reprioritize GH-244 if you'd like to get us up to 3.12.